src/requirements/courses-json-generator.ts

```diff
@@ -32,6 +32,14 @@ const getSubjects = async (semester: string): Promise<readonly string[]> => {
   }
 };
 
+const cleanString = (value: string | null) => {
+  if (value === null) return '';
+  return value.replace(/\u00a0/g, ' ');
+};
+
+const cleanMaybeString = (value: string | null | undefined): string | undefined =>
+  value?.replace(/\u00a0/g, ' ') || undefined;
```

Comment:
This is kind of confusing -- I think the only difference between including and not including `|| undefined` is for when value is `''`, in which case this function returns `undefined` rather than `''`. I don't know if that's the behavior we want. Let me know if you agree with the approach I suggested; then we can remove this weird detail.

Reply:
I only really looked at this briefly, but if we turn all undefineds into '', then we include extra information in the JSON (lots of empty strings). if we return undefined, then the field seems to be excluded from the JSON, which saves space. I'm just trying to preserve the current behavior… there's a lot that could be rethought with this though.

Reply:
Are empty strings returned from the api? In that case something like this kind of makes sense

Reply:
the API just doesn't return anything, I think. I'm only making the longTitle into an empty string because our code expects it to be non null. (also it's very weird that longTitle came in as null but other stuff is just excluded and I guess left as undefined… what's the difference between excluded / not present and undefined?)

Reply:
Probably just a buggy or undefined behavior in the api. Honestly I'm inclined to just keep it as `''` if it comes in as `''`.

Reply:
I don't think it does come in as `''`… I think there just isn't any information for certain stuff (like catalogComments or whatever). We really shouldn't add a '' for those because it makes the JSON bigger.

Reply:
If there is no data (i.e. the field is null or undefined), then it is redundant to include `|| undefined`